hub/api/dataset.py

```diff
@@ -310,15 +310,23 @@ def clear_cache(self):
         if hasattr(self.storage, "clear_cache"):
             self.storage.clear_cache()
 
-    def delete(self):
+    def __del__(self):
         """Deletes the entire dataset from the cache layers (if any) and the underlying storage.
-        This is an IRREVERSIBLE operation. Data once deleted can not be recovered.
+        This is an IRREVERSIBLE operation. Deleted data is not recoverable.
         """
```

Comment:
please don't use this, we determined __del__ would be terrible to have this functionality.

`__del__` is called upon reference death..

Reply:
please use the code from `release/2.0`